python/pyarrow/table.pxi

```diff
@@ -5358,36 +5358,45 @@ list[tuple(str, str, FunctionOptions)]
         ----
         values_sum: [[3,7,5]]
         keys: [["a","b","c"]]
+        >>> t.group_by("keys").aggregate([([], "count_all")])
+        pyarrow.Table
+        _count_all: int64
+        keys: string
+        ----
+        _count_all: [[2,2,1]]
+        keys: [["a","b","c"]]
         >>> t.group_by("keys").aggregate([])
         pyarrow.Table
         keys: string
         ----
         keys: [["a","b","c"]]
         """
-        columns = [a[0] for a in aggregations]
+        target_cols = [a[0] if isinstance(a[0], (list, tuple)) else [
+            a[0]] for a in aggregations]
```

Comment:
Why is this change needed? Because now the target cols are always expected to be a list?

Reply:
Looking further at the below changes, maybe we can combine all this extraction/validation in the single `for aggr in aggrfuncs:` loop below?  
That might give more readable code (compared to several of those list comprehensions), something like:

```
group_by_aggrs = []
for aggr in aggregations:
    if len(aggr) == 2:
        target, func = aggr
        opt = None
    elif:
        target, func, opt = aggr
    # check func name ...
    # convert target to list if needed ...
    group_by_aggrs.append((target, func, opt))
```